ananse/network.py

```diff
@@ -703,7 +713,12 @@ def run_network(
             logger.info("Writing network")
             out_dir = os.path.abspath(os.path.dirname(outfile))
             os.makedirs(out_dir, exist_ok=True)
-            result[["tf_target", "prob"]].to_csv(outfile, sep="\t", index=False)
+            if self.necessary_output:
+                logger.info("Writing only necesary prob score ")
+                result[["tf_target",  "prob"]].to_csv(outfile, sep="\t", index=False)
+            else:
+                logger.info("Writing all GRN variables prob network")
+                result[["tf_target","prob","tf_expression", "target_expression", "weighted_binding","activity"]].to_csv(outfile, sep="\t", index=False)
         else:
             return result[["tf_target", "prob"]]
```

Comment:
this bit can also use your argument
```
if self.necessary_output:
    return result[["tf_target",  "prob"]]
else:
    return result[["tf_target","prob","tf_expression", "target_expression", "weighted_binding","activity"]]
```

Reply:
"this bit" being line 723

Reply:
Line 723 is for outputing the GRN without generating a output file. I guess you could also store additional info in the object, but i think the use case where that would be needed is rather rare. :)